Mathlib/Data/Sum/Order.lean

```diff
@@ -333,6 +333,32 @@ theorem not_inr_le_inl [LE α] [LE β] {a : α} {b : β} : ¬toLex (inr b) ≤ t
 theorem not_inr_lt_inl [LT α] [LT β] {a : α} {b : β} : ¬toLex (inr b) < toLex (inl a) :=
   lex_inr_inl
 
+/-- `toLex` promoted to a `RelIso` between `<` relations. -/
+def toLexRelIsoLT [LT α] [LT β] :
+    Sum.Lex (· < · : α → α → Prop) (· < · : β → β → Prop) ≃r (· < · : α ⊕ₗ β → _ → _) :=
+  RelIso.refl _
+
+@[simp]
+theorem toLexRelIsoLT_apply [LT α] [LT β] (x : α ⊕ β) : toLexRelIsoLT x = toLex x :=
+  rfl
```

Comment:
```suggestion
@[simp]
theorem toLexRelIsoLT_coe [LT α] [LT β] : ⇑(toLexRelIsoLT (α := α) (β := β)) = ⇑toLex :=
  rfl
```
Maybe we want these theorems instead? (Same for the other 3.)

Reply:
These seem strictly more general, so sure.